### DISCUSSION TITLE
Dependency issues

### DISCUSSION THREAD

**Discussion by ziyaaktas** [ragapp/ragapp#204] [Q&A]
Getting dependency issues when running the app through dev container.  Wondering if it's normal or not.  Thanks.

**Comment by marcusschiesser** [comment]
can you provide us more details? e.g. what did you do and what was down?

**Comment by ziyaaktas** [comment]
Was getting errors like these below during `poetry install`
Because llama-index-vector-stores-chroma (0.1.9) depends on llama-index-core (>=0.10.1,<0.11.0)
 and llama-index (0.11.6) depends on llama-index-core (>=0.11.6,<0.12.0), llama-index-vector-stores-chroma (0.1.9) is incompatible with llama-index (0.11.6).
So, because app depends on both llama-index (0.11.6) and llama-index-vector-stores-chroma (0.1.9), version solving failed.
node ➜ .../src/ragapp/create_llama/backend $ poetry install
Skipping virtualenv creation, as specified in config file.
Updating dependencies
Resolving dependencies... (0.5s)

Because llama-index-vector-stores-chroma (0.1.10) depends on llama-index-core (>=0.10.1,<0.11.0)
 and llama-index (0.11.6) depends on llama-index-core (>=0.11.6,<0.12.0), llama-index-vector-stores-chroma (0.1.10) is incompatible with llama-index (0.11.6).
So, because app depends on both llama-index (0.11.6) and llama-index-vector-stores-chroma (0.1.10), version solving failed.

I changed a few versions, and I think it's ok now.

Using the dev-container flow btw.

**Comment by leehuwuj** [comment]
I believe the minimum version for llama-index-vector-stores-chroma needs to be `^0.2.0`, so `0.1.9` won't work.
(see: https://github.com/ragapp/ragapp/blob/e7e789d372d062ce654cd12fa109c253699853bb/src/ragapp/pyproject.toml#L41)
